### PR TITLE
Mhv 45263 compose form updates

### PR DESCRIPTION
**Pull request by KyleCardwell**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

1.  "Edit List" should be a button, not an underlined link
2.  ~~Width of text should span the length of the boxes--do we have a pixel or character #?~~
     - 2 was discussed with UCD and not able to be changed
3.  ~~All labels should be bold~~
     - 3 was discussed with UCD and not able to be changed
4.  Button width needs adjusting--do we have a universal size?
5.  Bottom buttons need new layout and icons (Add delete draft button)
6.  ~~Message Subject should be the next appropriate heading level (now H3, needs to be H2)~~
      - _6 was fixed in a different PR (https://github.com/department-of-veterans-affairs/vets-website/pull/24784/files)_

## Related issue(s)

- https://jira.devops.va.gov/browse/MHV-45263

## Testing done

- Old behavior: Delete draft button was not always visible
- New behavior: Delete draft button is visible before a draft is saved but clicking on it does nothing until the draft is saved
- Buttons realign at different breakpoints (desktop vs mobile)
- Unit / Cypress testing

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |Screenshot 2023-07-19 at 8 23 37 AM|Screenshot 2023-07-19 at 8 53 08 AM|
| Desktop |Screenshot 2023-07-19 at 8 23 07 AM|Screenshot 2023-07-19 at 8 52 29 AM|

## What areas of the site does it impact?

Va.gov - Secure Messaging

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 1]
```diff
@@ -254,7 +253,7 @@ const ComposeForm = props => {
   const saveDraftHandler = async (type, e) => {
     if (type === 'manual') {
       setUserSaved(true);
-      setLastFocusableElement(e.target.shadowRoot.querySelector('button'));
+      setLastFocusableElement(e.target);
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:256]: Perhaps worth validating that this focus lands properly with this change

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 2]
```diff
@@ -484,29 +482,35 @@ const ComposeForm = props => {
...
-            secondary
-            class="vads-u-flex--1 save-draft-button vads-u-margin-bottom--1"
+            className={`usa-button usa-button-secondary save-draft-button vads-u-flex--1 vads-u-margin-top--0 ${
+              draft ? 'vads-u-margin-right--1' : ''
+            }`}
             data-testid="Save-Draft-Button"
             onClick={e => saveDraftHandler('manual', e)}
+          >
+            <i className="fas fa-save" />
+            &nbsp;Save draft
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:506]: I am guessing you are trying to achieve some spacing with `&nbsp` here. Would a margin to an icon be more appropriate?

File: src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx [hunk 3]
```diff
@@ -388,24 +385,29 @@ const ReplyForm = props => {
...
-                  text="Save draft"
-                  secondary
-                  class="vads-u-flex--1 save-draft-button vads-u-margin-bottom--1"
+                  className="usa-button usa-button-secondary save-draft-button vads-u-flex--1 vads-u-margin-top--0 vads-u-margin-right--1"
                   data-testid="Save-Draft-Button"
                   onClick={e => saveDraftHandler('manual', e)}
-                />
+                >
+                  <i className="fas fa-save" />
+                  &nbsp;Save draft
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx:409]: same as a comment above

File: src/applications/mhv/secure-messaging/components/Draft/DeleteDraft.jsx [hunk 4]
```diff
@@ -40,17 +42,22 @@ const DeleteDraft = props => {
...
-          setIsModalVisible(true);
-          props.setLastFocusableElement(e.target);
+          if (props.draftId) {
+            setIsModalVisible(true);
+            props.setLastFocusableElement(e.target);
+          }
         }}
-      />
+      >
+        <i className="fas fa-trash-alt" />
```
[the remark this answers is not in the document]
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/components/Draft/DeleteDraft.jsx:58]: we are usually adding `aria-hidden` to icons

File: src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx [hunk 5]
```diff
@@ -388,24 +385,29 @@ const ReplyForm = props => {
...
                   id="save-draft-button"
-                  text="Save draft"
-                  secondary
-                  class="vads-u-flex--1 save-draft-button vads-u-margin-bottom--1"
+                  className="usa-button usa-button-secondary save-draft-button vads-u-flex--1 vads-u-margin-top--0 vads-u-margin-right--1"
                   data-testid="Save-Draft-Button"
                   onClick={e => saveDraftHandler('manual', e)}
-                />
+                >
+                  <i className="fas fa-save" />
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ReplyForm.jsx:408]: we are usually adding `aria-hidden` to icons

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 6]
```diff
@@ -484,29 +482,35 @@ const ComposeForm = props => {
...
-            text="Save draft"
-            secondary
-            class="vads-u-flex--1 save-draft-button vads-u-margin-bottom--1"
+            className={`usa-button usa-button-secondary save-draft-button vads-u-flex--1 vads-u-margin-top--0 ${
+              draft ? 'vads-u-margin-right--1' : ''
+            }`}
             data-testid="Save-Draft-Button"
             onClick={e => saveDraftHandler('manual', e)}
+          >
+            <i className="fas fa-save" />
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:505]: we are usually adding `aria-hidden` to icons

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]

**Review by alex-morgun-1 — APPROVED** [review]

**Review by vsaleem — APPROVED** [review]

**Pull request merged**